### ISSUE TITLE
Fade in / Fade out animation buggy

### ISSUE THREAD

**Issue by crtlf** [OwlCarousel2/OwlCarousel2#2319]
- jQuery : 1.12.4
- Owl Carousel : 2.3.3

```javascript
$('.details-slideshow, .list-carousel').owlCarousel({
  items: 1,
  loop: false,
  nav: false,
  dots: true,
  rewind: true,
  autoplay: true,
  autoplayTimeout: 8000,
  autoplayHoverPause: true,
  checkVisible: false,
  animateOut: 'fadeOut',
  animateIn: 'fadeIn'
});
```

When using this configuration, my slides sometimes disappears (I really don't understand why or when it occurs), because the `div.owl-item` keeps some fade animation classes (see below).

#### When it works

Default behavior.

```html
<div class="owl-stage">
  <div class="owl-item"></div>
  <div class="owl-item active"></div>
</div>
```

#### After a while, it gets buggy

The fade animation classes (`animated owl-animated-out fadeOut owl-animated-in fadeIn`) stay in one or multiple items, causing them to stay invisible.

```html
<div class="owl-stage">
  <div class="owl-item animated owl-animated-out fadeOut owl-animated-in fadeIn"></div>
  <div class="owl-item animated owl-animated-out fadeOut owl-animated-in fadeIn active"></div>
</div>
```

The big problem of this is I don't understand how this bug appears, I don't have anything to do to make it happen, apart letting the carousel run a few seconds or minutes.

**Comment by crtlf** [comment]
Ok, I think I've sort it out, even though I don't really understand why.

First thing, I had completely missed out the animate.css library in order to use the "fadeIn/fadeOut" animation. I don't know how, but it was working without it. At first, I thought this was causing the bug, but no.

Finally, I _think_ I managed to make it work simply by changing `loop` from `false` to `true`. Maybe the carousel gets buggy when you have only 2 items fading into each other.

Sorry if I didn't put a jsFiddle, but I couldn't reproduce the bug (that made it even more difficult to solve it).

**Comment by crtlf** [comment]
I don't know why but it's still bugging, maybe less often, but still.

Here is a live version of the website : https://www.brest-terres-oceanes.fr
It's the big carousel on top of the page. If you're lucky, you'll see a slide losing his content by keeping all his animation classes `animated owl-animated-out fadeOut owl-animated-in fadeIn`.

**Comment by iamrobert** [comment]
Yes - I am encountering it to on another site. However I don't see it on this one.

The steps are:
1. Click the 3 dots rather than scroll 
2. Click the first dot again

DId you manage to fix it?

**Comment by crtlf** [comment]
I don't, unfortunately.

The only thing I manage to do is decreasing the number of times the bug appears, by changing the `loop` from `false` to `true`. Maybe because it clones the carousel items so it's more difficult to trigger the bug. With only 2 items, and with `loop` to `false`, it seems easier to trigger it.

**Comment by iamrobert** [comment]
I made a demo - however it doesn't have a problem:

https://codepen.io/iamrobert/pen/gjyLYo

So maybe - its something else.

**Comment by iamrobert** [comment]
I have a working demo but not sure how its working. Also - it removes the transition - but I am animating my slides - so didn't want the slide effect.

I added a translate event to remove the classeses.

```javascript
onTranslate: function (event) {
	$('.owl-item').removeClass('owl-animated-out');
	$('.owl-item').removeClass('fadeOut');
	$('.owl-item').removeClass('animated');
},
```

Then I commented out position: relative:
```css
  .owl-carousel .owl-item {
// position: relative;
    min-height: 1px;
    float: left;
    -webkit-backface-visibility: hidden;
    -webkit-tap-highlight-color: transparent;
    -webkit-touch-callout: none; }
```

**Comment by trolleyweb** [comment]
I had the same issue and I think I figured out why. Were you using the animateIn/animateOut settings? I had the following settings:
```
animateIn: 'fadeIn',
animateOut: 'fadeOut'
```
OwlCarousel would start out fine, then begin adding the class `fadeOut` to the active item and `fadeIn` to the inactive item.  This would cause the active item to be invisible. I didn't figure out the root of the issue but I removed the animate settings and that solved my use-case.

**Comment by Deesen** [comment]
I think I have an idea what is going on. When using custom CSS3-Animations inside a slide like fading in/out different text-layers, ownCarousel will not remove classes "owl-animated-out fadeOut animated" like mentioned already by @iamrobert, when slides are changed and custom animations not finished yet. So I used

```
         onTranslated: function (event) {
            setTimeout(function(){
                $('.owl-item')
                    .removeClass('owl-animated-in')
                    .removeClass('owl-animated-out')
                    .removeClass('fadeIn')
                    .removeClass('fadeOut')
                    .removeClass('animated');
            },1500);
        }
```
to make sure those classes are removed after my custom animations are finished. Issues are solved completely for me this way... if not I will mention here.
